src/components/StrongPasswordInput/index.js

```diff
@@ -0,0 +1,248 @@
+        const { bottomHelpText } = props;
+        if (bottomHelpText) {
+            return inlineTextLabelId;
+        }
+        return undefined;
+    }
+
+    function getErrorMessageId() {
+        const { error } = props;
+        if (error) {
+            return errorMessageId;
+        }
+        return undefined;
+    }
+
+    const {
+        id,
+        className,
+        style,
+        bottomHelpText,
+        error,
+        passwordState,
+        passwordStateLabel,
+        value,
+        placeholder,
+        onChange,
+        disabled,
+        readOnly,
+        tabIndex,
+        onFocus,
+        onBlur,
+        onClick,
+        onKeyDown,
+        maxLength,
+        minLength,
+        pattern,
+        icon,
+        required,
+        autoComplete,
+        name,
+        label,
+        hideLabel,
+        isBare,
+        isCentered,
+        iconPosition,
+    } = props;
+
+    let status;
+    if (passwordState === 'poor') {
+        status = 'error';
+    } else if (passwordState === 'average') {
+        status = 'warning';
+    } else if (passwordState === 'strong') {
+        status = 'success';
+    }
+
+    return (
+        <StyledContainer id={id} className={className} style={style}>
+            <Label
+                label={label}
+                hideLabel={hideLabel}
+                required={required}
+                inputId={inputId}
+                readOnly={readOnly}
+                id={getInlineTextLabelId()}
+            />
+
+            <RelativeElement>
+                <RenderIf isTrue={!!icon}>
+                    <StyledIconContainer
+                        iconPosition={iconPosition}
+                        readOnly={readOnly}
+                        error={error}
+                    >
+                        {icon}
+                    </StyledIconContainer>
+                </RenderIf>
+
+                <StyledInput
+                    id={inputId}
+                    name={name}
+                    type="password"
+                    value={value}
+                    placeholder={placeholder}
+                    onChange={onChange}
+                    tabIndex={tabIndex}
+                    onFocus={onFocus}
+                    onBlur={onBlur}
+                    onClick={onClick}
+                    onKeyDown={onKeyDown}
+                    disabled={disabled}
+                    readOnly={readOnly}
+                    required={required}
+                    maxLength={maxLength}
+                    minLength={minLength}
+                    pattern={pattern}
+                    autoComplete={autoComplete}
+                    aria-labelledby={getInlineTextLabelId()}
+                    aria-describedby={getErrorMessageId()}
+                    isBare={isBare}
+                    isCentered={isCentered}
+                    iconPosition={iconPosition}
+                    icon={icon}
+                    error={error}
```

Comment:
They will not be doubled because I am not reusing the Input component as it is necessary to render the status bar between the input and the help text and change the input border color, and there is no available mechanism to achieve that, so I could only reuse the StyledInput from Input/styled and had to duplicate all the other functionality

Reply:
I didn't realize that, I thought you were reusing the Input component